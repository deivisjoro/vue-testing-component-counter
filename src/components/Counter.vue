<script setup>
    import { computed, ref } from 'vue';

    const props = defineProps({
        inicio: {
            default: 1,
            type: Number
        }
    })

    const counter = ref(props.inicio);

    const incrementar = () => {
        counter.value++;
    }

    const decrementar = () => {
        counter.value--;
    }

    const squareCounter = computed(()=>{
        return counter.value * counter.value
    })

</script>

<template>
    <h2>Counter App</h2>
    <div>
        <h3>{{ counter }} <sup>2</sup> = {{ squareCounter }} </h3>
        <p data-testid="counter">{{ counter }}</p>
        <div class="buttons">
            <button @click="decrementar" data-testid="button-decrementar"> - </button>
            <button @click="incrementar" data-testid="button-incrementar"> + </button>
        </div>
    </div>
  
</template>

<style scoped>
    button{
        padding: 3px 5px;
        margin-right: 10px;
    }
</style>