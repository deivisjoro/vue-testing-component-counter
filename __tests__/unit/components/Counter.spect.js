import { beforeEach, describe, expect, test } from 'vitest';
import { shallowMount } from '@vue/test-utils'
import Counter from '@/components/Counter.vue';

describe('Counter Component', () => {
    
    // const wrapper = shallowMount(Counter);
    let wrapper;
    beforeEach(()=>{
        wrapper = shallowMount(Counter);
    })
  
    test('Debe ser mayor a 10', ()=>{
        let value = 11;
        
        expect(value).toBeGreaterThan(10);
    })

    test('Debe hacer match con el snapshot', () => {
        const wrapper = shallowMount(Counter);

        expect(wrapper.html()).toMatchSnapshot();
    })

    test('Render tag H2 and text content is "Counter App"', () => {
        const h2 = wrapper.find('h2');
        expect(h2.exists()).toBeTruthy();
        expect(h2.text()).toBe('Counter App');
    })

    test('debe incrementar y decrementar el valor del contador', async () => { 
        
        const button1 = wrapper.find('[data-testid="button-incrementar"]');
        expect(button1.exists()).toBeTruthy();
        await button1.trigger('click');
        const counterContainer = wrapper.find('[data-testid="counter"]');
        expect(counterContainer.exists()).toBeTruthy();
        let value = counterContainer.text();
        expect(value).toBe("2");

        const button2 = wrapper.find('[data-testid="button-decrementar"]');
        expect(button2.exists()).toBeTruthy();
        await button2.trigger('click');
        await button2.trigger('click');
        value = counterContainer.text();
        expect(value).toBe("0");

    })

    test('debe establecer el valor por defecto', ()=>{
        // const { inicio } = wrapper.props();
        // const inicio = wrapper.props('inicio');
        // const inicio = wrapper.props().inicio;
        const inicio = wrapper.props()['inicio'];

        const counterContainer = wrapper.find('[data-testid="counter"]');
        expect(counterContainer.exists()).toBeTruthy();
        const value = counterContainer.text();
        expect(Number(value)).toBe(inicio);
    })

    test('debe establecer el valor enviado por props', ()=>{
        const testValue = 10;
        wrapper = shallowMount(Counter, {
            props: {
                inicio: testValue
            }
        });
        const { inicio } = wrapper.props();
        const counterContainer = wrapper.find('[data-testid="counter"]');
        expect(counterContainer.exists()).toBeTruthy();
        const value = counterContainer.text();
        expect(Number(value)).toBe(inicio);
        expect(Number(value)).toBe(testValue);
    })

})
