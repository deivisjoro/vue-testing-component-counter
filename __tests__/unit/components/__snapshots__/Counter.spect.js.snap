// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Counter Component > Debe hacer match con el snapshot 1`] = `
"<h2 data-v-9bd91872=\\"\\">Counter App</h2>
<div data-v-9bd91872=\\"\\">
  <h3 data-v-9bd91872=\\"\\">1 <sup data-v-9bd91872=\\"\\">2</sup> = 1</h3>
  <p data-v-9bd91872=\\"\\" data-testid=\\"counter\\">1</p>
  <div data-v-9bd91872=\\"\\" class=\\"buttons\\"><button data-v-9bd91872=\\"\\" data-testid=\\"button-decrementar\\"> - </button><button data-v-9bd91872=\\"\\" data-testid=\\"button-incrementar\\"> + </button></div>
</div>"
`;
